packages/cli/src/controllers/auth.controller.ts

```diff
@@ -194,16 +193,13 @@ export class AuthController {
 		return { loggedOut: true };
 	}
 
-	private async validateMfaToken(user: User, token?: string) {
-		if (!!!token) return false;
-		return this.mfaService.totp.verifySecret({
-			secret: user.mfaSecret ?? '',
-			token,
-		});
+	private async validateMfaToken(token: string | undefined, secret: string) {
```

Comment:
Would it make sense to type the `secret` as a special type so we avoid confusion when passing encrypted vs decrypted versions of it? Right now it's all strings so it's easy to confuse.

Reply:
I'm working on a second PR to refactor all this code, improve typing, and add more tests (if possible)